src/views/ApartmentApp/index.js

```diff
@@ -74,60 +144,64 @@ export default class ApartApp extends Component {
         this.setState({ applicants });
       }
     }
+    this.setState({ updating: false });
   };
 
   /**
-   * Loads the user's profile info only once (at start)
+   * Callback for apartment application save button
    */
-  async loadProfile() {
-    this.setState({ loading: true });
-    try {
-      const profile = await user.getProfileInfo();
-      this.setState({ userProfile: profile });
-      this.checkPersonType(profile);
-      if (this.state.isStu) {
-        let applicants = this.state.applicants;
-        applicants.push(profile);
-        this.setState({ applicants });
-      }
-      this.setState({ loading: false });
-    } catch (error) {
-      // Do Nothing
-    }
-  }
-
-  checkPersonType(profile) {
-    let personType = String(profile.PersonType);
-    this.setState({ isStu: personType.includes('stu') });
-    this.setState({ isFac: personType.includes('fac') });
-    this.setState({ isAlu: personType.includes('alu') });
-  }
+  handleSaveButtonClick = () => {
+    // The method is separated from callback because the housing API service must be handled inside an async method
+    let debugMessage = 'DEBUG: Save button was clicked';
+    console.log(debugMessage);
+    // alert(debugMessage);
+    this.saveApplication(this.state.userProfile.AD_Username, this.state.applicants);
+  };
 
   /**
-   * Loads the user's saved apartment application, if one exists
+   * Save the current state of the application to the database
+   * @param {String} primaryUsername the student username of the person filling out the application
+   * @param {StudentProfileInfo} applicants Array of StudentProfileInfo objects
    */
-  async loadHousingInfo() {
-    this.setState({ loading: true });
+  async saveApplication(primaryUsername, applicants) {
+    this.setState({ saving: true });
+    this.saveButtonAlertTimeout = null;
+    let saveSuccessful = null;
     try {
-      // TODO - Once saving application has been implemented in the backend, this will be replaced with a call to the load the application info. The getHousingInfo was made obsolete after the Hello World
-      let housingInfo = await housing.getHousingInfo();
-      let onOffCampus = String(housingInfo[0].OnOffCampus);
-      let onCampusRoom = String(housingInfo[0].OnCampusRoom);
-      this.setState({ onOffCampus, onCampusRoom, loading: false });
+      const saved = await housing.saveApartmentApplication(primaryUsername, applicants);
+      console.log(saved.primaryUsername); // `saved` is a placeholder name
+      saveSuccessful = saved;
     } catch (error) {
       // Do Nothing
     }
+    if (saveSuccessful) {
+      this.setState({ saving: 'success' });
```

Comment:
Do you want to use the state 'savingSuccess' and 'true' here? (and set it to false on 183?)

Reply:
`this.state.savingSuccess` was deprecated and I forgot to remove it, which I have now done